tipsy/TipViewController.swift

```diff
@@ -48,39 +92,4 @@ class ViewController: UIViewController {
         tipLabel.text = String(format: "$%.2f", tip)
         totalLabel.text = String(format: "$%.2f", total)
     }
-    
-    override func viewDidAppear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        
-        let defaults = UserDefaults.standard
-        let intValue = defaults.object(forKey: "default_tip_index")
-        var timeAppClosed = defaults.object(forKey: "time_app_closed") as? Int
-        if (timeAppClosed == nil) {
-            timeAppClosed = 0
-        }
-        let currentTime = Int(NSDate().timeIntervalSince1970)
-        if (intValue != nil) {
-            tipControl.selectedSegmentIndex = intValue as! Int
-            updateTip()
-        }
-        if (currentTime - timeAppClosed! < 600) {
-            billField.text = defaults.object(forKey: "bill_last_closed") as? String
-            bill = Double(billField.text!) ?? 0
```

Comment:
I really try to avoid using `!`. It's just asking your app to crash if `nil` occurs for that variable at some point. I probably unwrap optionals excessively but I like to air on the side of caution.